vit-servicing-station-cli/src/csv/models.rs

```diff
@@ -137,11 +140,48 @@ impl From<Proposal> for proposals::Proposal {
             chain_vote_encryption_key: proposal.chain_vote_encryption_key,
             fund_id: proposal.fund_id,
             challenge_id: proposal.challenge_id,
-            proposal_solution: proposal.proposal_solution,
-            proposal_brief: proposal.proposal_brief,
-            proposal_importance: proposal.proposal_importance,
-            proposal_goal: proposal.proposal_goal,
-            proposal_metrics: proposal.proposal_metrics,
+        }
+    }
+}
+
+impl TryFrom<Proposal> for ProposalChallengeInfo {
+    type Error = std::io::Error;
+
+    fn try_from(p: Proposal) -> Result<Self, Self::Error> {
+        let values = (
+            p.proposal_solution,
+            p.proposal_brief,
+            p.proposal_importance,
+            p.proposal_goal,
+            p.proposal_metrics,
+        );
+
+        match values {
+            (Some(proposal_solution), None, None, None, None) => {
+                Ok(ProposalChallengeInfo::Simple(SimpleChallengeProposal {
+                    proposal_solution,
+                }))
+            }
+            (
+                None,
+                Some(proposal_brief),
+                Some(proposal_importance),
+                Some(proposal_goal),
+                Some(proposal_metrics),
+            ) => Ok(ProposalChallengeInfo::Community(CommunityChallengeProposal {
```

Comment:
Since we don't have the information available of the type in the proposal itself we can match set attributes here.
It may be wiser to check against the challenge itself for data sanity, but It will also make it more of a heavy load computation. Should we ensure this, or do we trust data input enough to skip it?

Reply:
I'd rather use challenge type from the challenge struct, if we can associate it at this point.

Reply:
We do not have a relationship build at that time. But we can build it using a hashmap or finding in the challenge array before this.